paasta_tools/cli/cmds/spark_run.py

```diff
@@ -606,6 +636,30 @@ def get_spark_app_name(original_docker_cmd: Union[Any, str, List[str]]) -> str:
     return spark_app_name
 
 
+def _calculate_docker_memory_cpu_limit(spark_conf):
+    try:
+
+        docker_memory_limit_str = spark_conf.get(
+            "spark.driver.memory", DEFAULT_DRIVER_MEMORY_BY_SPARK
+        )
+        match = re.match(r"([0-9]+)([a-z]*)", docker_memory_limit_str)
+        memory_val = int(match[1]) * DOCKER_RESOURCE_ADJUSTMENT_FACTOR
+        memory_unit = match[2]
+        docker_memory_limit = f"{memory_val}{memory_unit}"
+    except Exception as e:
+        # For any reason it fails, continue with default value
+        log.error(
+            f"Failed to parse docker memory limit. Error: {e}. Example values: 1g, 200m."
+            " Setting a default value {DEFAULT_MAX_DOCKER_MEMORY_LIMIT}"
+        )
+        docker_memory_limit = DEFAULT_MAX_DOCKER_MEMORY_LIMIT
+
+    docker_cpu_limit = str(
+        int(spark_conf.get("spark.driver.cores", DEFAULT_DRIVER_CORES_BY_SPARK))
+    )
```

Comment:
Do we ensure spark.driver.cores is convertible to an int anywhere before this, or should we catch exceptions here too like we do for memory?

Reply:
1. If spark.driver.cores is convertible to int before this, it would still work, right?
2. I tested in manual test cases, and didn't see any issues. Ticket comments for results
3. I thought about catching the error here too, but this seems like to basic operation. Only cases I see it failing is spark.driver.cores is provided as (non-int) string or float (or a different type than int-str). 
Happy to catch it for extra caution.

Reply:
One more note about this is that `docker run` will flat out fail to run if we specify more `--cpus` than the host has total -- with our batch boxes having only 8vcpu, we'd want to be a little careful about this; some configs in yelpsoa-configs already appear to specify spark.driver.cores=8, thankfully none more than 8.

Is there already documentation to tell folks not to use >8? If not, the user experience might be slightly unfriendly, where they only find out their tronjob's config is not valid when the job fails.

Reply:
Good Catch. I tested this max value case for driver-memory (in the ticket) where in `docker run` just set the value to max memory of the box by default. Strange to me, that it didn't do the same for `cpus' too.
I will test the same for the max value too.

What should be the max value you suggest? This is considering that many instance types are being used for batch and stable_batch boxes?
Is 8vcpu's standard and reliable long-term consistent value to be used to check for max allowed value?

Reply:
8 seems fine to leave as the max although it's literally the maximum available on these hosts. The batch and stable_batch boxes aren't involved with the driver, and the boxes that run our drivers are all only 1 instance type right now w/ only 8vcpu. 

So since we're counting on drivers moving to k8s soon (where we can actually schedule based on resources requested), it might be sufficient just to leave this and update documentation, maybe add a pre-commit to yelpsoa-configs.

Reply:
Ok. I will leave it like that. 
Interesting in local test, when i tried setting 10 cores. It just let me and continued running.
`docker inspect 6c5e647b4632 | grep Nano
            "NanoCpus": 10000000000,`
Maybe it becomes more of the useless value when greater than max-cores as opposed to an issue. 
Which I think might be ok.

Reply:
^This test was done on a host that had >10 cpu available; can you try again either specifying more cpu than your test host has or testing on one of the batch boxes?

my concern here is that for most `spark-run` tron jobs, the driver will be run on a host that has only 8 cpu available, so we either a) give users a bad experience where they can set up yelpsoa-configs with >8 spark.driver.cores for their job, then only find out it fails on the first scheduled run, or b) add some sort of safety here to just set this value to maximum for the host it's running on

Reply:
I tried with > 10 cpus. It failed. I added more details in the help section for better user experience. This has been discussed offline with @jfongatyelp